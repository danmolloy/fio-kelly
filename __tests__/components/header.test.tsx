import "@testing-library/jest-dom"
import { act, fireEvent, render, screen } from "@testing-library/react"
import Header, { HeaderProps, menuItems } from "../../components/header"
import "react-scroll"
import { Link, animateScroll } from "react-scroll";

jest.mock("react-scroll", () => ({
  animateScroll: {
    scrollToTop: jest.fn()
  },
  Link: (props) => <div></div>
}))


const mockProps: HeaderProps = {
  showMenu: false,
  showHeader: true,
  setShowHeader: jest.fn(),
  setShowMenu: jest.fn(),
}

describe("<Header />", () => {
  beforeEach(() => {
    render(<Header {...mockProps} />)
  })
  it("page-header is in the document", () => {
    const pageHeader = screen.getByTestId("page-header")
    expect(pageHeader).toBeInTheDocument()
  })
  it("'Fiona Kelly' title is in the document", () => {
    const titleText = screen.getByText("Fiona Kelly")
    expect(titleText).toBeInTheDocument()
  })
  it("'Home' button is in the document and calls scrollToTop on click", async () => {
    const homeBtn = screen.getByText("Home")
    expect(homeBtn).toBeInTheDocument()
    await act(async () => {
      await fireEvent.click(homeBtn)
    })
    expect(animateScroll.scrollToTop).toHaveBeenCalledWith({
      smooth: true, 
      duration: 500
    })
  })
  it("<MenuIcon /> is in the document", () => {
    const menuIcon = screen.getByTestId("menu-icon")
    expect(menuIcon).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const pageHeader = screen.getByTestId("page-header")
    expect(pageHeader).toMatchSnapshot()
  })
  //it("all menu buttons are in the document and calls scroll on click", () => {})
})