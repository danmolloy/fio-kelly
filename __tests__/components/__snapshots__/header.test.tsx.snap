// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<div
  class="font-display  transition duration-500 ease-out fixed fill-white  z-30 h-16 w-screen top-0 flex flex-row  items-center justify-between   text-zinc-100"
  data-testid="page-header"
>
  <h1
    class="md:hidden p-4 "
  >
    Fiona Kelly
  </h1>
  <div
    class="hidden md:flex w-screen  flex-row  justify-end pr-4"
  >
    <button
      class=" p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400"
    >
      Home
    </button>
    <div />
    <div />
    <div />
    <div />
  </div>
  <button
    class="false m-2 hover:fill-amber-500 active:fill-amber-400  md:hidden"
    data-testid="menu-icon"
  >
    <svg
      class="m-2"
      height="22"
      width="30"
    >
      <rect
        height="2"
        rx="2"
        style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
        width="30"
      />
      <rect
        height="2"
        rx="2"
        style="opacity: 1;"
        width="30"
        y="8"
      />
      <rect
        height="2"
        rx="2"
        style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
        width="30"
        y="17"
      />
    </svg>
  </button>
</div>
`;
